frontend/src/pages/user/UserAnswerPage.js

```diff
@@ -39,13 +41,48 @@ export default function UserAnswerPage() {
   const lastPage =
     questions && Array.isArray(questions) ? questions.length + 1 : 0;
 
+  const totalQuestion =
+    questions && Array.isArray(questions) ? questions.length : 0;
+  const passingScore = 75;
+
+  useEffect(() => {
+    setPassed(scorePercentage >= passingScore ? true : false);
+  }, [scorePercentage]);
+
+  useEffect(() => {
+    const calculate = async () => {
+      try {
+        const result = await calculateScore({
+          score,
+          totalQuestion,
+          answers,
+          correctAnswers,
+        });
+        setScorePercentage(result.result);
+        setScore(result.score);
+        setIsCorrect(result.isCorrect);
+      } catch (error) {
+        console.log(error);
+      }
+    };
+    calculate();
+  }, [isDone]);
```

Comment:
1. You can set right away if passed or not, dont create useEffect and another variable. its taking a memory
2. rename it to response. Its not good to use result.result.
3. Again, much better to calculate the `passed` in BE so pls remove this calculation `response.result >= passingScore ? true : false`in FE. You are doing the calculation in BE already why not do it all there?
4. What is the `isCorrect` variable used for?Seems like its referring to a single question. Maybe rename it like `result`


```suggestion

  useEffect(() => {
    const calculate = async () => {
      try {
        const response = await calculateScore({
          score,
          totalQuestion,
          answers,
          correctAnswers,
        });
        setScorePercentage(response.result);
        setPassed(response.result >= passingScore ? true : false);
        setScore(response.score);
        setIsCorrect(response.isCorrect);
      } catch (error) {
        console.log(error);
      }
    };
    calculate();
  }, [isDone]);
```

Reply:
isCorrect is I use to pass on AnswerResultPage to render the status if isCorrect then show correct image, else show wrong image.